data/publiccloud/azure_runcmd.sh

```diff
@@ -0,0 +1,145 @@
+cmd_status "az_vm_run-command_invoke" az vm run-command invoke \
+    --command-id RunShellScript \
+    --resource-group "$grpname" \
+    --name "$vmname" \
+    --scripts "cat /etc/waagent.conf"
+
+
+echo "Create Run-command $runcmd { az vm run-command create } "
+cmd_status "az_vm_run-command_create" az vm run-command create --name "$runcmd" -g "$grpname" --vm-name "$vmname" --script "echo Agent-status-ready'"
+
+echo "List created Run-command $runcmd { az vm run-command list } "
+echo "*************************************************************"
+cmd_status "az_vm_run-command_list" az vm run-command list --vm-name "$vmname" -g "$grpname" --location "$location"
+for cmdlist in $(az vm run-command list --vm-name $vmname -g $grpname --location $location --query "[].{name:name}" -o tsv); do 
+    cmd_status "az_vm_run-command_show" az vm run-command show -o table --vm-name "$vmname" -g "$grpname" --run-command-name "$cmdlist" --location "$location"
+done
+
+echo "Delete Run-command $runcmd { az vm run-command delete } "
+echo "*************************************************************"
+cmd_status "az_vm_run-command_delete" az vm run-command delete -g "$grpname" --run-command-name "$runcmd" --vm-name "$vmname" --yes -y
+
+echo "Delete vm and the associated resources like { az resource list } "
+echo "*************************************************************"
+cmd_status "az_resource_list" az resource list -o table -g "$grpname"
+az resource list -o table -g $grpname --query "[?contains(name, 'oqacliruncmdvm')].{Name:name,Type:type}" -o tsv
+
+#Get required ids to delete the resource
+echo "Get interface id for the network resource { az vm show --query } "
+echo "*************************************************************"
+cmd_status "az_vm_show_querynetworkinterface" az vm show --resource-group "$grpname" --name "$vmname"
+interface_id=$(az vm show --resource-group $grpname --name $vmname --query networkProfile.networkInterfaces[0].id)
+interface_id=${interface_id:1: -1}
+
+echo "Get os disk id for the managedDisk { az vm show --query } "
+echo "*************************************************************"
+cmd_status "az_vm_show_querymanagedDisk" az vm show --resource-group "$grpname" --name "$vmname"
+os_disk_id=$(az vm show --resource-group ${grpname} --name ${vmname} --query storageProfile.osDisk.managedDisk.id)
+os_disk_id=${os_disk_id:1: -1}
+
+echo "Get security group id { az vm show --query } "
+echo "*************************************************************"
+cmd_status "az_vm_show_querysecgrp" az vm show --resource-group "$grpname" --name "$vmname"
+security_group_id=$(az network nic show --id ${interface_id} --query networkSecurityGroup.id)
+security_group_id=${security_group_id:1: -1}
+
+echo "Get public ip id { az vm show --query } "
+echo "*************************************************************"
+cmd_status "az_vm_show_publicip" az vm show --resource-group "$grpname" --name "$vmname"
```

Comment:
`az vm show --resource-group "$grpname" --name "$vmname"` - can you call it once , save to the variable and than show a variable ?

Reply:
I can store the vm show in a variable and grep all the required details. 
Instead I want to call cli with different parameter combination.